src/client/coll.rs

```diff
@@ -0,0 +1,194 @@
+use bson;
+use client::db::Database;
+use client::common::{ReadPreference, WriteConcern};
+use client::wire_protocol::operations::{OpQueryFlags, Message};
+
+/// Interfaces with a MongoDB collection.
+pub struct Collection<'a> {
+    db: &'a Database<'a>,
+    pub name: String,
+    pub namespace: String,
```

Comment:
It seems rather superfluous having "name" and "namespace" Strings represent a Collection. Within the server, we have a lightweight NamespaceString object (a full-qualified namespace for a collection like "test.first_coll" = <db>.<collection>). The name of a collection can easily be gotten from the NamespaceString.